contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,448 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol" as Ownable;
+
+contract SeriesIndexer is Ownable.Ownable {
+    // Counter
+    uint256 private nextSeriesID = 1;
+    uint256 private nextArtistID = 1;
+
+    struct Series {
+        string metadata;
+        string contractTokenData;
+        uint256[] artistIDs;
+    }
+
+    // Artist Management
+    mapping(uint256 => address) private artistIDToAddress;
+    mapping(address => uint256) private addressToArtistID;
+    mapping(uint256 => bool) private ownerRightsRevokedForArtistID;
+
+    // Mappings for series and artists
+    mapping(uint256 => Series) private seriesDetails;
+    mapping(uint256 => uint256[]) private artistIDSeriesIDs;
+    mapping(uint256 => mapping(uint256 => bool)) private isArtistIDInSeries;// SeriesID => ArtistID => Bool
+    mapping(uint256 => mapping(uint256 => bool)) private seriesPendingCoArtist;// SeriesID => ArtistID => Bool
+
+    // Pending Requests Management
+    mapping(uint256 => uint256[]) private artistPendingCoArtistRequests;// artistID => SeriesID[]
+    mapping(uint256 => mapping(uint256 => uint256)) private artistPendingCoArtistRequestIndex;// artistID => SeriesID => Index
+
+    // Events
+    event ArtistAddressUpdated(uint256 indexed artistID, address indexed oldAddress, address indexed newAddress);
+    event SeriesIndexed(
+        uint256 indexed seriesID,
+        uint256[] artistIDs,
+        string metadata,
+        string seriesContractTokenCID
```

Comment:
I would prefer using something like `URI` instead of `CID` that we need to stick to ipfs. Using `URI` could be better for extensibility later on.